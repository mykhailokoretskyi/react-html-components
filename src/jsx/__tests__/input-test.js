jest.mock('react/lib/ReactDefaultInjection');
import React from 'react';
import InputBox from '../inputBox.jsx';
import renderer from 'react/lib/ReactTestRenderer';

const textInputWithListeners = (
    <InputBox required={true}
           label="test label"
           name="test_input"
           id="test"
           placeholder="Input"
           type="text"
           extraClass="anotherClass"
           validate={true}
           errorMessage="wrong"
           successMessage="correct"
           changeCallback={function(){}}
           mouseEnterCallback={function(){}}
           mouseLeaveCallback={function(){}}/>
);

const textInputWithoutListeners = (
    <InputBox required={true}
           label="test label"
           name="test_input"
           id="test"
           type="text"/>
);

const textInputWithValue = (
    <InputBox label="test label"
           name="test_input"
           id="test"
           value="value"
           type="text"/>
);

describe('Input', () => {
    it('with listeners matches snapshot', () => {
        const component = renderer.create(textInputWithListeners);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('without listeners matches snapshot', () => {
        const component = renderer.create(textInputWithoutListeners);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('with value matches snapshot', () => {
        const component = renderer.create(textInputWithValue);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
