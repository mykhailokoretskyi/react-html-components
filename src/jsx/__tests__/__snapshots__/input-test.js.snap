exports[`Input with listeners matches snapshot 1`] = `
<div
  className="input-field col s12">
  <input
    className="validate anotherClass"
    disabled={false}
    id="test"
    name="test_input"
    onChange={[Function bound _onChange]}
    onMouseEnter={[Function bound _onMouseEnter]}
    onMouseLeave={[Function bound _onMouseLeave]}
    placeholder="Input"
    required={true}
    type="text"
    value="" />
  <label
    dangerouslySetInnerHTML={
      Object {
        "__html": "test label"
      }
    }
    data-error="wrong"
    data-success="correct"
    htmlFor="test" />
</div>
`;

exports[`Input with value matches snapshot 1`] = `
<div
  className="input-field col s12">
  <input
    className=""
    disabled={false}
    id="test"
    name="test_input"
    onChange={[Function bound _onChange]}
    onMouseEnter={[Function bound _onMouseEnter]}
    onMouseLeave={[Function bound _onMouseLeave]}
    placeholder=""
    required={false}
    type="text"
    value="value" />
  <label
    dangerouslySetInnerHTML={
      Object {
        "__html": "test label"
      }
    }
    data-error=""
    data-success=""
    htmlFor="test" />
</div>
`;

exports[`Input without listeners matches snapshot 1`] = `
<div
  className="input-field col s12">
  <input
    className=""
    disabled={false}
    id="test"
    name="test_input"
    onChange={[Function bound _onChange]}
    onMouseEnter={[Function bound _onMouseEnter]}
    onMouseLeave={[Function bound _onMouseLeave]}
    placeholder=""
    required={true}
    type="text"
    value="" />
  <label
    dangerouslySetInnerHTML={
      Object {
        "__html": "test label"
      }
    }
    data-error=""
    data-success=""
    htmlFor="test" />
</div>
`;
